docs/src/stream.rst

```diff
@@ -146,8 +146,11 @@ API
 
 .. c:function:: int uv_read_stop(uv_stream_t*)
 
-    Stop reading data from the stream. The :c:type:`uv_read_cb` callback will
-    no longer be called.
+    Stop reading data from the stream.
+
+    If libuv still owns memory from an earlier :c:type:`uv_alloc_cb` call, it
+    invokes the :c:type:`uv_read_cb` call once with `nread == 0` to give back
+    the memory. No further :c:type:`uv_read_cb` callbacks will be made.
```

Comment:
The way this is written does not make any promises about how long libuv might hold on to the allocation.  Is that intentional?

It seems like libuv is very close to the guarantee of "read_stop means no more read_cb", with one exception where a user does something pretty weird (call read_stop in an alloc_cb but return an allocation anyway).  It would be nice to offer that guarantee to users, even with the exception, because a user can keep a simpler mental model of how their code will behave.

Reply:
This makes sense... am I reading the libuv code correctly, that this only happens in `src/win/tty.c`?

Reply:
Looks like it to me also. But seems like we can change that to use `ReadConsoleInputExW` with the `NOWAIT` flag now, and avoid the considerable associated complexity (e.g. e51442bbc9b37dff32430c047844d9181798a9a8)

Reply:
Even with `NOWAIT` I wouldn't want to tighten the guarantees. For one, that would preclude moving to io_uring; it wants you to commit the memory upfront.

(_Technically_ you don't have to but at that point it's just a clunkier epoll.)